### ISSUE TITLE
About logging LPS TWR data

### ISSUE THREAD

**Issue by easy98** [USC-ACTLab/crazyswarm#283]
Hi,

I am currently setting up the LPS system based on #108 using the TWR mode. My purpose is to get the LPS raw data, which seems to be the distance between the tag and anchors [here](https://github.com/bitcraze/crazyflie-firmware/blob/13aa1372bc3ba5b14922fc2f9d8999270882021a/src/deck/drivers/src/lpsTwrTag.c#L569-#L593). I would like to directly log it in [this line](https://github.com/USC-ACTLab/crazyswarm/blob/ab0fe73a05b6e947d5651c411634eb49a3862efc/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L13).  
But I am not sure if it's the right way to log that data and if it's possible to to that? Also,I noticed that my position estimation is not so correct based on [here](https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch#14). If this state estimation is predicted according to this [estimator_kalmen](https://github.com/bitcraze/crazyflie-firmware/blob/12cf59a535a859c260b00d0b558681be911e759a/src/modules/src/estimator_kalman.c)?

Thanks in advance.

**Comment by whoenig** [comment]
If you want to log the data in real-time over the radio, this is the correct approach. All you need to do is figuring out the logging variable of interest - in your case `ranging.distance0` etc. If you just want to look at the data in post-processing, I recommend using the uSD-card deck instead (you will be able to log more variables at a higher rate compared to the over-the-radio mode).

If you use the LPS, the Kalman filter is always active automatically, so you should be getting the state as predicted by the on-board filter.

**Comment by easy98** [comment]
It works! Thanks Whoenig！

**Comment by easy98** [comment]
Hi, 

When I switched the mode to TDOA3 on the python client, I changed the logging variable to "tdoa3.tdoa". But when I run "roslaunch crazyswarm hover_swarm.launch", It throws an error saying “Could not find tdoa3.tdoa in log toc!”. Then I rechanged the logging variable to "ranging.distance0", no error occurred but the value of "ranging.distance0" was always zero and the mode was actually TDOA3 rather than TWR. I am not sure why I am facing this error.
